<a href="https://colab.research.google.com/github/marioespinosa-sys/MatBach_Python/blob/main/MetodoBabol%C3%B3nico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import math

n = float(input("Dame el valor del cual quieres la raíz: "))
aprox = float(input("Dam una aproximación: "))

sucRaiz = []

for i in range(6):
  aprox = (1/2)*(aprox + n/aprox)
  sucRaiz.append(aprox)

  print("La raíz aproximada de {0} es {1:.20}".format(n,aprox))
  print()
  print("La raíz con math de {0} es {1:.20}".format(n,math.sqrt(n)))
  print()
  print("La suceción es: ", sucRaiz)
  print()

Dame el valor del cual quieres la raíz: 45
Dam una aproximación: 7
La raíz aproximada de 45.0 es 6.7142857142857144126

La raíz con math de 45.0 es 6.7082039324993694152

La suceción es:  [6.714285714285714]

La raíz aproximada de 45.0 es 6.7082066869300911449

La raíz con math de 45.0 es 6.7082039324993694152

La suceción es:  [6.714285714285714, 6.708206686930091]

La raíz aproximada de 45.0 es 6.7082039324999342966

La raíz con math de 45.0 es 6.7082039324993694152

La suceción es:  [6.714285714285714, 6.708206686930091, 6.708203932499934]

La raíz aproximada de 45.0 es 6.7082039324993694152

La raíz con math de 45.0 es 6.7082039324993694152

La suceción es:  [6.714285714285714, 6.708206686930091, 6.708203932499934, 6.708203932499369]

La raíz aproximada de 45.0 es 6.7082039324993694152

La raíz con math de 45.0 es 6.7082039324993694152

La suceción es:  [6.714285714285714, 6.708206686930091, 6.708203932499934, 6.708203932499369, 6.708203932499369]

La raíz aproximada de 45.0 es 6.7

Actividad:
Ver con cuantas iteraciones y con cuántas cifras se aproxima correctamente para:
a) raíz de 11 con aproximación 3

In [12]:
import math

n = float(input("Dame el valor del cual quieres la raíz: "))
aprox = float(input("Dam una aproximación: "))

sucRaiz = []

for i in range(3):
  aprox = (1/2)*(aprox + n/aprox)
  sucRaiz.append(aprox)

  print("La raíz aproximada de {0} es {1:.20}".format(n,aprox))
  print()
  print("La raíz con math de {0} es {1:.20}".format(n,math.sqrt(n)))
  print()
  print("La suceción es: ", sucRaiz)
  print()
  if abs(aprox - math.sqrt(n)) < 0.0000001:
    print("El valor obtenido es muy cercano al valor obtenido con math")
    print("Estas iteraciones son suficientes para que la solución se considere correcta")
  else:
    print("El valor obtenido no es muy cercano al valor obtenido con math")
    print("La diferencia entre los valores es de {0:.20}".format(abs(aprox - math.sqrt(n))))
  print()



Dame el valor del cual quieres la raíz: 11
Dam una aproximación: 3
La raíz aproximada de 11.0 es 3.3333333333333330373

La raíz con math de 11.0 es 3.31662479035539981

La suceción es:  [3.333333333333333]

El valor obtenido no es muy cercano al valor obtenido con math
La diferencia entre los valores es de 0.016708542977933227291

La raíz aproximada de 11.0 es 3.3166666666666664298

La raíz con math de 11.0 es 3.31662479035539981

La suceción es:  [3.333333333333333, 3.3166666666666664]

El valor obtenido no es muy cercano al valor obtenido con math
La diferencia entre los valores es de 4.1876311266619836715e-05

La raíz aproximada de 11.0 es 3.3166247906197652284

La raíz con math de 11.0 es 3.31662479035539981

La suceción es:  [3.333333333333333, 3.3166666666666664, 3.3166247906197652]

El valor obtenido es muy cercano al valor obtenido con math
Estas iteraciones son suficientes para que la solución se considere correcta



Actividad:
Ver con cuantas iteraciones y con cuántas cifras se aproxima correctamente para:
b) raíz de 2 con aproximación 1

In [14]:
import math

n = float(input("Dame el valor del cual quieres la raíz: "))
aprox = float(input("Dam una aproximación: "))

sucRaiz = []

for i in range(1):
  aprox = (1/2)*(aprox + n/aprox)
  sucRaiz.append(aprox)
  dif = abs(aprox - math.sqrt(n))
  print("La raíz aproximada de {0} es {1:.20}".format(n,aprox))
  print()
  print("La raíz con math de {0} es {1:.20}".format(n,math.sqrt(n)))
  print()
  print("La suceción es: ", sucRaiz)
  print()
  if abs(aprox - math.sqrt(n)) < 0.0000001:
    print("El valor obtenido es muy cercano al valor obtenido con math")
    print("Estas iteraciones son suficientes para que la solución se considere correcta")
  else:
    print("El valor obtenido no es muy cercano al valor obtenido con math")
    print("Estas iteraciones no son suficientes para que la solución se considere correcta")
    print("La diferencia entre los valores es de {0:.20}".format(dif))
  print()

Dame el valor del cual quieres la raíz: 2
Dam una aproximación: 1
La raíz aproximada de 2.0 es 1.5

La raíz con math de 2.0 es 1.4142135623730951455

La suceción es:  [1.5]

El valor obtenido no es muy cercano al valor obtenido con math
Estas iteraciones no son suficientes para que la solución se considere correcta
La diferencia entre los valores es de 0.085786437626904854525



Actividad:
Ver con cuantas iteraciones y con cuántas cifras se aproxima correctamente para:
b.1) raíz de 2 con aproximación 4

In [24]:
import math

n = float(input("Dame el valor del cual quieres la raíz: "))
aprox = float(input("Dam una aproximación: "))

sucRaiz = []

for i in range(4):
  aprox = (1/2)*(aprox + n/aprox)
  sucRaiz.append(aprox)
  dif = abs(aprox - math.sqrt(n))
  print("La raíz aproximada de {0} es {1:.20}".format(n,aprox))
  print()
  print("La raíz con math de {0} es {1:.20}".format(n,math.sqrt(n)))
  print()
  print("La suceción es: ", sucRaiz)
  print()
  if abs(aprox - math.sqrt(n)) < 0.0000001:
    print("El valor obtenido es muy cercano al valor obtenido con math")
    print("Estas iteraciones son suficientes para que la solución se considere correcta")
    print("La diferencia entre los valores es de {0:.20}".format(dif))
  else:
    print("El valor obtenido no es muy cercano al valor obtenido con math")
    print("Estas iteraciones no son suficientes para que la solución se considere correcta")
    print("La diferencia entre los valores es de {0:.20}".format(dif))
  print()

Dame el valor del cual quieres la raíz: 231
Dam una aproximación: 15
La raíz aproximada de 231.0 es 15.199999999999999289

La raíz con math de 231.0 es 15.198684153570663824

La suceción es:  [15.2]

El valor obtenido no es muy cercano al valor obtenido con math
Estas iteraciones no son suficientes para que la solución se considere correcta
La diferencia entre los valores es de 0.0013158464293354654728

La raíz aproximada de 231.0 es 15.198684210526316463

La raíz con math de 231.0 es 15.198684153570663824

La suceción es:  [15.2, 15.198684210526316]

El valor obtenido es muy cercano al valor obtenido con math
Estas iteraciones son suficientes para que la solución se considere correcta

La raíz aproximada de 231.0 es 15.198684153570663824

La raíz con math de 231.0 es 15.198684153570663824

La suceción es:  [15.2, 15.198684210526316, 15.198684153570664]

El valor obtenido es muy cercano al valor obtenido con math
Estas iteraciones son suficientes para que la solución se considere corre

Actividad.
Ver con cuantas iteraciones y con cuatas cifras se aproxima correctamente para:
c) raíz de 231 poniendo el prámetro de formato del número a 49 cifras decimales

In [27]:
import math

n = float(input("Dame el valor del cual quieres la raíz: "))
aprox = float(input("Dam una aproximación: "))

sucRaiz = [60]

for i in range(1):
  aprox = (1/2)*(aprox + n/aprox)
  sucRaiz.append(aprox)
  dif = abs(aprox - math.sqrt(n))
  print("La raíz aproximada de {0} es {1:.49}".format(n,aprox))
  print()
  print("La raíz con math de {0} es {1:.49}".format(n,math.sqrt(n)))
  print()
  print("La suceción es: ", sucRaiz)
  print()
  if abs(aprox - math.sqrt(n)) < 0.0000001:
    print("El valor obtenido es muy cercano al valor obtenido con math")
    print("Estas iteraciones son suficientes para que la solución se considere correcta")
    print("La diferencia entre los valores es de {0:.20}".format(dif))
  else:
    print("El valor obtenido no es muy cercano al valor obtenido con math")
    print("Estas iteraciones no son suficientes para que la solución se considere correcta")
    print("La diferencia entre los valores es de {0:.20}".format(dif))
  print()

Dame el valor del cual quieres la raíz: 231
Dam una aproximación: 15
La raíz aproximada de 231.0 es 15.19999999999999928945726423989981412887573242188

La raíz con math de 231.0 es 15.19868415357066382398443238344043493270874023438

La suceción es:  [60, 15.2]

El valor obtenido no es muy cercano al valor obtenido con math
Estas iteraciones no son suficientes para que la solución se considere correcta
La diferencia entre los valores es de 0.0013158464293354654728

